tests/unit/plugins/module_utils/test_api_module.py

```
@@ -142,3 +148,288 @@ def test_basic_auth(mock_api, api_module_with_user_pass):
+def test_delete_auth_provider(api_module):
+    mock_api_instance = MagicMock()
+    mock_api_instance.delete_auth_provider.return_value = None
+
+    with patch.dict('plugins.module_utils.constants.API_MAPPING', {
+        ResourceType.AUTH_PROVIDER: MagicMock(
+            api=MagicMock(return_value=mock_api_instance),
+            api_version='v1beta1',
+            delete='delete_auth_provider',
+        ),
+    }):
+        result = api_module.delete(ResourceType.AUTH_PROVIDER, "test-provider", None)
```

Flake8 (7.3.0)

[error] local variable 'result' is assigned to but never used

(F841)